Add next cell buttons

garyb opened the pull request:
@cryogenian Here's what I was thinking might help with the issue you have in #565

The "next cell" buttons now only appear when the results cell actually has an output value, so from that we should know for sure that a child will be autorun-able as soon as it is added.

I guess maybe we need to add some weird thing to actually solve the problem for `Query` though: perhaps running a query cell with no input does not actually run a query, but instead can just be used to push the latest `VarMap` in for autocomplete purposes?

cryogenian commented:
We can make cell with `Maybe Port` argument and use `initializer`.

cryogenian commented on wip/src/Notebook/Cell/Component/State.purs:
```diff
@@ -89,6 +92,8 @@ type CellState =
   , hasResults :: Boolean
   , cachingEnabled :: Maybe Boolean -- Nothing if the option isn't available
+  , lastInput :: Maybe Port
+  , lastOutput :: Maybe Port
```
This makes `Notebook.Component.State.values` redundant. We must contain input/output consistency anyway.

garyb replied:
Yeah, I know :smile: - that's why this is still WIP, just wanted your thoughts first.

cryogenian replied:
So, yeah, :+1: for removing `values` and using `input/output`. It works in master and we haven't got any problems with it.

cryogenian commented on wip/src/Notebook/Cell/Component/State.purs:
```diff
@@ -105,6 +110,8 @@ initEditorCellState accessType visibility =
   , messageVisibility: Invisible
   , hasResults: false
   , cachingEnabled: Nothing
+  , lastInput: Nothing
```
Oh, and why not name them just `input` and `output`? :)

garyb commented:
Updated to remove `values` stuff now.

Also fixed a bug in autorunning of cells added via `CreateChildCell` that crept in at some point - it was assuming the new cell would be `cellId + one`, but that's only true if we were adding a child cell at the end of the notebook.

cryogenian commented:
Thanks! :+1:
Travis fails because of `apt-get update`.

garyb commented:
Yeah, I saw that, thanks. We're not testing `new-halogen` yet anyway so I wasn't too concerned :smile:
